### PR TITLE
fix(eslint-plugin): fix false positive from adjacent-overload-signatures

### PR DESCRIPTION
**Pull request by lukyth**
`adjacent-overload-signatures` rule won't distinguish between static and instance methods, despite they're different. So when they have the same name, they'll be treated as the same thing, which can cause a false positive result.

Fix https://github.com/typescript-eslint/typescript-eslint/issues/169

### CODE REVIEW

**Comment by bradzacher** [comment]
thanks for the submission! Solution looks good!

I think in this case, a better solution could be to switch the interim variable's type entirely to an object.
That way you're avoiding the string concatenation cost, your code is more explicit, and it leaves it open for us to support more modifiers in future, if the need arrises.

```
lastMember = {
  name: 'str',
  static: true,
}
```

**Comment by lukyth** [comment]
@bradzacher Thank you so much for your comment 😄 I've refactored the code according to it (2285b6c). Could you please check that out?

File: packages/eslint-plugin/lib/rules/adjacent-overload-signatures.js [hunk 1]
```diff
@@ -84,27 +95,34 @@ module.exports = {
 
       if (members) {
         let name;
```
[the remark this answers is not in the document]
Comment (lukyth) [reply packages/eslint-plugin/lib/rules/adjacent-overload-signatures.js]: Only `lastMethod` and `seenMethods` are being reused. So I'll move the rest (`name`, `method`, and `index`) into the `forEach` scope.

**Comment by lukyth** [comment]
@bradzacher Done :)

**Pull request merged**